service/resource/persistentvolume/resource.go

```diff
@@ -133,3 +136,100 @@ func pvToRecyclePV(v interface{}) (*RecyclePersistentVolume, error) {
 
 	return rpv, nil
 }
+
+// newRecycleStateAnnotation create new PersistentVolume object with
+// updated recycle state annotation.
+func (r *Resource) newRecycleStateAnnotation(pv *apiv1.PersistentVolume, recycleAnnotation string) (*apiv1.PersistentVolume, error) {
+
+	updatedpv := &apiv1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        pv.Name,
+			Annotations: pv.Annotations,
+			Labels: pv.Labels,
+		},
+		Spec: apiv1.PersistentVolumeSpec{
+			Capacity:                      pv.Spec.Capacity,
+			StorageClassName:              pv.Spec.StorageClassName,
+			AccessModes:                   pv.Spec.AccessModes,
+			PersistentVolumeReclaimPolicy: pv.Spec.PersistentVolumeReclaimPolicy,
+			PersistentVolumeSource:        pv.Spec.PersistentVolumeSource,
+		},
+	}
+
+	r.logger.Log("persistentvolume", pv.Name, "set new recycle annotation", recycleAnnotation)
+	updatedpv.ObjectMeta.Annotations[recycleStateAnnotation] = recycleAnnotation
+	
+	return updatedpv, nil
+}
+
+// newPvc returns k8s PersistentVolumeClaim object,
+// which bounds persistent volume from function parameter.
+func newPvc(pv *apiv1.PersistentVolume) *apiv1.PersistentVolumeClaim {
+
+	pvc := &apiv1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("pv-cleaner-claim-%s", pv.Name),
+			Namespace: "kube-system",
+		},
+		Spec: apiv1.PersistentVolumeClaimSpec{
+			StorageClassName: &pv.Spec.StorageClassName,
+			AccessModes:      pv.Spec.AccessModes,
+			Resources: apiv1.ResourceRequirements{
+				Requests: pv.Spec.Capacity,
+			},
+			VolumeName: pv.Name,
+		},
+	}
+
+	return pvc
+}
+
+// newCleanupJob returns k8s job objects,
+// which runs busybox container, mounts claim from the function parameter
+// and run shell command to cleanup mount path.
```

Comment:
<3

Reply:
Can you put some more explanation of that `test -e /scrub && rm -rf /scrub/..?* /scrub/.[!.]* /scrub/*  && test -z \"$(ls -A /scrub)\" || exit 1 ` in the godoc, for ppl like me who don't understand :)

Reply:
Actually I took it from official documentation: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#recycling